### PR TITLE
Deprecate `confirmTransaction`, `getSignatureStatus`, and `getSignatureConfirmation`

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Various client JSON RPC libraries appear to have problems processing the `status: { "Ok": null } `transaction status returned by methods such as `getSignatureStatuses`. The `null` value causes the removal of `Ok` entirely and the client ends up with a status object of `{}`.

We don't know who is using these two endpoints so we'll have to support them for now, but we should actively discourage anyone from using them from this point forward.

#### Summary of Changes
* Remove all uses of these endpoints from our clients
* Remove docs
* Mark the endpoints as deprecated in the code

Related to: https://github.com/solana-labs/solana/issues/9297

### CODE REVIEW

File: client/src/rpc_client.rs [hunk 1]
```diff
@@ -77,17 +77,16 @@ impl RpcClient {
...
-        let response = self
-            .client
-            .send(
-                &RpcRequest::ConfirmTransaction,
-                json!([signature.to_string(), commitment_config]),
-                0,
-            )
-            .map_err(|err| err.into_with_command("ConfirmTransaction"))?;
+        let Response { context, value } =
+            self.get_signature_statuses(&[*signature], commitment_config)?;
```
Comment (CriesofCarrots) [inline client/src/rpc_client.rs:81]: Update method name as per suggestion below

File: client/src/rpc_client.rs [hunk 2]
```diff
@@ -113,6 +112,21 @@ impl RpcClient {
         self.get_signature_status_with_commitment(signature, CommitmentConfig::default())
     }
 
+    pub fn get_signature_statuses(
```
[the remark this answers is not in the document]
Comment (mvines) [reply client/src/rpc_client.rs]: fixed

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

File: core/src/rpc.rs [hunk 3]
```diff
@@ -589,24 +608,8 @@ pub trait RpcSol {
...
-    fn get_signature_status(
-        &self,
-        meta: Self::Metadata,
-        signature_str: String,
-        commitment: Option<CommitmentConfig>,
-    ) -> Result<Option<transaction::Result<()>>>;
-
     #[rpc(meta, name = "getSignatureStatuses")]
-    fn get_signature_statuses(
+    fn get_signature_statuses_with_commitment(
```
Comment (CriesofCarrots) [inline core/src/rpc.rs:612]: I only intended this name change in rpc_client. This is actually not consistent with the rest of the rpc methods. Will revert as part of transaction-history work.

**Pull request merged**